Mathlib/RingTheory/MvPowerSeries/LinearTopology.lean

```diff
@@ -0,0 +1,203 @@
+/-
+Copyright (c) 2024 Antoine Chambert-Loir, María Inés de Frutos-Fernández. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Chambert-Loir, María Inés de Frutos-Fernández
+-/
+import Mathlib.Data.Finsupp.Interval
+import Mathlib.RingTheory.MvPowerSeries.PiTopology
+import Mathlib.Topology.Algebra.LinearTopology
+import Mathlib.Topology.Algebra.Nonarchimedean.Bases
+import Mathlib.RingTheory.TwoSidedIdeal.Operations
+
+/-! # Linear topology on the ring of multivariate power series
+
+- `MvPowerSeries.basis`: the ideals of the ring of multivariate power series
+all coefficients the exponent of which is smaller than some bound vanish.
+
+## Instances :
+
+- `MvPowerSeries.isLinearTopology` : if `R` has a linear topology,
+then the product topology on `MvPowerSeries σ R` is a linear topology.
+This applies in particular when `R` has a discrete topology.
+
+-/
+
+namespace MvPowerSeries
+
+namespace LinearTopology
+
+open scoped Topology
+
+open Set SetLike
+
+/-- The underlying family for the basis of ideals in a multivariate power series ring. -/
+def basis (σ : Type*) (R : Type*) [Ring R] (Jd : TwoSidedIdeal R × (σ →₀ ℕ)) :
+    TwoSidedIdeal (MvPowerSeries σ R) := by
+  apply TwoSidedIdeal.mk' {f | ∀ e ≤ Jd.2, coeff R e f ∈ Jd.1}
+  · simp [coeff_zero]
+  · exact fun hf hg e he ↦ by rw [map_add]; exact add_mem (hf e he) (hg e he)
+  · exact fun {f} hf e he ↦ by simp only [map_neg, neg_mem, hf e he]
+  · exact fun {f g} hg e he ↦ by
+      classical
+      rw [coeff_mul]
+      apply sum_mem
+      rintro uv huv
+      apply TwoSidedIdeal.mul_mem_left
+      exact hg _ (le_trans (le_iff_exists_add'.mpr
+        ⟨uv.fst, (Finset.mem_antidiagonal.mp huv).symm⟩) he)
```

Comment:
Am I right that this can be simplified using [Finset.antidiagonal.fst_le](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Order/Antidiag/Prod.html#Finset.antidiagonal.fst_le) or [Finset.antidiagonal.snd_le](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Order/Antidiag/Prod.html#Finset.antidiagonal.snd_le)?

Reply:
yep

Reply:
and the next one too.